### ISSUE TITLE
SOTA Model for Text Prompt Segmentation

### ISSUE THREAD

**Issue by xiaobanni** [facebookresearch/segment-anything#575]
I am looking for a state-of-the-art (SOTA) model for **text prompt** segmentation. Currently, I am aware of two choices: [Grounded-Segment-Anything](https://github.com/IDEA-Research/Grounded-Segment-Anything) and [SEEM](https://github.com/UX-Decoder/Segment-Everything-Everywhere-All-At-Once). However, both of these models fail to meet my requirements.

Consider the following example: I want the model to segment the lane lines, but the results from the aforementioned methods are as follows (i hope they can segment the lane line in the road):

Grounded-Segment-Anything:
<img width="1175" alt="image" src="https://github.com/facebookresearch/segment-anything/assets/33119031/89883c7f-d5de-41ae-89d0-a79f10680a17">

SEEM Model:
<img width="1507" alt="image" src="https://github.com/facebookresearch/segment-anything/assets/33119031/9c724f67-643f-4071-92fa-acf61cdead98">

Unfortunately, neither of them can solve this problem effectively. I would greatly appreciate any recommendations you may have.

Any information regarding the timeline for the release of SAM text-prompt capabilities would be welcome.

**Comment by emi-research-dl** [comment]
I recommend you this: https://github.com/luca-medeiros/lang-segment-anything

**Comment by xiaobanni** [comment]
Thank you for the recommendation. However, I have tried it and found that it is just an easier-to-read version of [Grounded-Segment-Anything](https://github.com/IDEA-Research/Grounded-Segment-Anything). It uses the same method of using GroundingDINO to translate the text prompt to a box prompt and then sending it to SAM, resulting in similar outcomes to the Grounded-Segment-Anything mentioned earlier. I believe that an oriented text prompt segment model (rather than the two-stage invoking) is necessary to address the issue at hand and facilitate broader downstream applications.

**Comment by TerryYiDa** [comment]
> I am looking for a state-of-the-art (SOTA) model for **text prompt** segmentation. Currently, I am aware of two choices: [Grounded-Segment-Anything](https://github.com/IDEA-Research/Grounded-Segment-Anything) and [SEEM](https://github.com/UX-Decoder/Segment-Everything-Everywhere-All-At-Once). However, both of these models fail to meet my requirements.
> 
> Consider the following example: I want the model to segment the lane lines, but the results from the aforementioned methods are as follows (i hope they can segment the lane line in the road):
> 
> Grounded-Segment-Anything: <img alt="image" width="1175" src="https://user-images.githubusercontent.com/33119031/270579943-89883c7f-d5de-41ae-89d0-a79f10680a17.png">
> 
> SEEM Model: <img alt="image" width="1507" src="https://user-images.githubusercontent.com/33119031/270579442-9c724f67-643f-4071-92fa-acf61cdead98.png">
> 
> Unfortunately, neither of them can solve this problem effectively. I would greatly appreciate any recommendations you may have.
> 
> Any information regarding the timeline for the release of SAM text-prompt capabilities would be welcome.

Do you have any good solutions? I'm facing the same problem now

**Comment by xiaobanni** [comment]
@TerryYiDa No. So, I hope this issue can track the progress of the advanced text-prompt segmentation model.

**Comment by iacopo97** [comment]
I have the same problem, do you find a solution?

**Comment by YuetianW** [comment]
Lol, really wish it was possible to open up the ability to use text prompts . A two-stage approach like Grounded-Segment-Anything is neither useful nor elegant.😣

**Comment by muhammadsr** [comment]
Anyone made progress with this issue?
